### ISSUE TITLE
Test fails randomly

### ISSUE THREAD

**Issue by aannleax** [knowsys/nemo#417]
Just had this error message:

```
---- io::formats::rdf_triples::test::example_1 stdout ----
thread 'io::formats::rdf_triples::test::example_1' panicked at nemo/src/io/formats/rdf_triples.rs:284:9:
assertion `left == right` failed
  left: "_:subje"
 right: "\"object1\""
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Only occurs once in like 20 runs.

**Comment by monsterkrampe** [comment]
Trying to reproduce, I'm also getting another failed test sometimes:
```
---- io::formats::dsv::test::csv_with_ignored_and_faulty stdout ----
thread 'io::formats::dsv::test::csv_with_ignored_and_faulty' panicked at nemo-physical/src/dictionary/hash_map_dictionary.rs:90:22:
index out of bounds: the len is 5 but the index is 11688260230290059526
```

**Comment by monsterkrampe** [comment]
The output in the original description also looks weird: `_:subje` looks very incomplete.
Seems to me like this could be related to the new dictionary (also because of the second failed test I found).

**Comment by monsterkrampe** [comment]
Found another one:
```
---- dictionary::hash_map_dictionary::test::fetch_id stdout ----
thread 'dictionary::hash_map_dictionary::test::fetch_id' panicked at nemo-physical/src/dictionary/hash_map_dictionary.rs:394:9:
assertion `left == right` failed
  left: None
 right: Some(0)
stack backtrace:
   0: rust_begin_unwind
             at /rustc/ba7c7a301984967c8c13adb580ef9b86ba706a83/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/ba7c7a301984967c8c13adb580ef9b86ba706a83/library/core/src/panicking.rs:72:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
             at /rustc/ba7c7a301984967c8c13adb580ef9b86ba706a83/library/core/src/panicking.rs:279:5
   4: nemo_physical::dictionary::hash_map_dictionary::test::fetch_id
             at ./src/dictionary/hash_map_dictionary.rs:394:9
   5: nemo_physical::dictionary::hash_map_dictionary::test::fetch_id::{{closure}}
             at ./src/dictionary/hash_map_dictionary.rs:392:18
   6: core::ops::function::FnOnce::call_once
             at /rustc/ba7c7a301984967c8c13adb580ef9b86ba706a83/library/core/src/ops/function.rs:250:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/ba7c7a301984967c8c13adb580ef9b86ba706a83/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by monsterkrampe** [comment]
As @mkroetzsch pointed out offline, this is due to multiple dictionaries using the same global string buffer to store data.
Since tests run in parallel, each test still gets its own dictionary but all of them modify the same string buffer...

**Comment by mkroetzsch** [comment]
To me more precise, several HashMapDictionary structs should never use the same string buffer pages, since page allocation is protected by a lock, as long as distinct threads use distinct dictionary objects. More analysis is needed to find out if the locking mechanism is insufficient or another issue might be the cause.
